a5py/routines/runmixin.py

```diff
@@ -517,15 +530,27 @@ def getwall_loads(self, weights=True):
         i0 = 0
         for i in range(wetted.size):
             i1 = idx[i] + 1
-            pdepo[i]  = np.sum(weight[i0:i1])
-            edepo[i]  = np.sum(energy[i0:i1]*weight[i0:i1])
+            if weights: #include weights in the deposition
+                pdepo[i]  = np.sum(weight[i0:i1])
+                edepo[i]  = np.sum(energy[i0:i1]*weight[i0:i1])
+            else:
+                pdepo[i]  = i1 - i0
+                edepo[i]  = np.sum(energy[i0:i1])
             iangle[i] = 0 #TBD
             i0 = i1
 
         return wetted, area, edepo, pdepo, iangle
 
-    def getwall_3dmesh(self):
+    def getwall_3dmesh(self, wall_filter = None, particle_filter = None):
         """Return 3D mesh representation of 3D wall and associated loads.
+        Inputs
+        -------
+            wall_filter : list[int], optional
+                List of triangle indecies for which the 3D mesh is made
+
+            particle_filter : list[int], optional
+                List of particle ids for which the wall loads are calculated
+
```

Comment:
No spaces in keyword arguments, Line change before "Inputs", wall_filter and particle_filter should have same indentation as Inputs. Yeah and periods. In Python docstrings sentences start with a capital letter and end in period.